helm-chart/binderhub/templates/ingress.yaml

```diff
@@ -23,12 +23,22 @@ spec:
               serviceName: binder
               servicePort: 8585
     {{- end }}
-{{- if and .Values.ingress.https.enabled (eq .Values.ingress.https.type "kube-lego") }}
+{{- if .Values.ingress.https.enabled }}
   tls:
+  {{- if eq .Values.ingress.https.type "kube-lego" }}
     - secretName: kubelego-tls-binder-{{ .Release.Name }}
       hosts:
         {{- range $host := .Values.ingress.hosts }}
         - {{ $host }}
         {{- end }}
+  {{- else }}
+    {{- range .Values.ingress.tls }}
+    - hosts:
```

Comment:
Not `{{ .Values.ingress.tls | toYaml | nindent 6 }}` ? That should avoid duplicating the structure here in the template.

Reply:
Not sure what you mean here; they do the same thing in the end. As I mentioned, I took the `ingress.yaml` from what's created using `helm create`, which is the closest to a standard you could get since it's in the official binary. The template in full is:

```yaml
{{- if .Values.ingress.enabled -}}
{{- $fullName := include "test-chart.fullname" . -}}
{{- $ingressPath := .Values.ingress.path -}}
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: {{ $fullName }}
  labels:
    app.kubernetes.io/name: {{ include "test-chart.name" . }}
    helm.sh/chart: {{ include "test-chart.chart" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- with .Values.ingress.annotations }}
  annotations:
{{ toYaml . | indent 4 }}
{{- end }}
spec:
{{- if .Values.ingress.tls }}
  tls:
  {{- range .Values.ingress.tls }}
    - hosts:
      {{- range .hosts }}
        - {{ . | quote }}
      {{- end }}
      secretName: {{ .secretName }}
  {{- end }}
{{- end }}
  rules:
  {{- range .Values.ingress.hosts }}
    - host: {{ . | quote }}
      http:
        paths:
          - path: {{ $ingressPath }}
            backend:
              serviceName: {{ $fullName }}
              servicePort: http
  {{- end }}
{{- end }}
```

And it's generated from Helm version:
```
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```

Reply:
No, it's fine if there's a reason. Just an improvement I thought I saw.